src/Ocelot/Configuration/Creator/RoutesCreator.cs

```diff
@@ -122,7 +122,7 @@ private DownstreamRoute SetUpDownstreamRoute(FileRoute fileRoute, FileGlobalConf
                 .WithClaimsToDownstreamPath(claimsToDownstreamPath)
                 .WithRequestIdKey(requestIdKey)
                 .WithIsCached(fileRouteOptions.IsCached)
-                .WithCacheOptions(new CacheOptions(fileRoute.FileCacheOptions.TtlSeconds, region, fileRoute.FileCacheOptions.Header))
+                .WithCacheOptions(new CacheOptions(fileRoute.FileCacheOptions.TtlSeconds, region, fileRoute.FileCacheOptions.Header, fileRoute.FileCacheOptions.EnableContentHashing))

```

Comment:
@raman-m maybe we should foresee a CacheOptions creator? We are already retrieving the region ```var region = _regionCreator.Create(fileRoute);```, why couldn't we deal with the other properties in a CacheOptions creator?

Reply:
Historically, Ocelot utilizes the design concept of `Creator` classes. Regarding copying constructors in C#, a constructor may accept parameters of any type. Employing a constructor can significantly reduce the amount of code written, as exemplified in line 125.

Reply:
Yes, I agree, but I was wondering if we could have a creator that would include the region creation too? So we wouldn't add too much lines of codes and we would group the cache options creation... The code is then easier to read.

Reply:
The decision should be left to the author. Personally, I favor constructors for simple initialization and member assignments. However, for complex logic, such as calling an interface object, it is preferable to define a "creator" class with dependencies.

Reply:
@raman-m Ok, so I decided to review this code since it seems to me that several users are now complaining about it. No feedback from the author until now... So I would recommend following the  creator strategy.  By the way, when the new cache has been implemented, at least one acceptance test verifying the EnableContentHashing option should have been foreseen, but that's not the case. I will add an acceptance test for that matter.

Reply:
> No feedback from the author until now... 

According to my statistics, approximately 9 out of 10 pull requests in this project from open-source contributors consist of one or two commits that express an idea but lack a solid understanding of how to implement the feature or fix. As a team, we must adapt to this situation and ensure the delivery is taken care of.
Sad, but true...

----

> So I would recommend following the creator strategy.

### 👌